### PR TITLE
Fix tabs extending past table

### LINKED ISSUE #734: Difficulty Tabs Extends Past Mods Table
**Describe the bug**
The difficulty tabs extend past the mods table when the screen gets too narrow. Also, the difficulty tabs stop aligning with the right side of the mods table when the horizontal scroll bar appears.

**To Reproduce**
1. Go to "/mods".
2. Make the window narrower until the effect appears.

**Expected behavior**
The difficulty tabs component should limit itself to the width of the mods table and hide any elements that extend past it. Also, the right side of the difficulty tabs should always align with the right side of the mods table.

**Screenshots**
image

**Additional context**
If this winds up being too difficult to fix with our code, we can fork [the datatable](https://github.com/icflorescu/mantine-datatable-v6) and modify it so that the tabs can be passed in and render within the datatable component. This would involve some more work down the road though when we eventually migrate to Mantine v7 (#599).

Also, if the right alignment is much more difficult to solve than the tabs extending outside of the table, then that can be left as a follow-up issue for a later milestone.

### PR DESCRIPTION
**Pull request by ShouvikGhosh2048**
Closes #734

Uses [portals](https://react.dev/reference/react-dom/createPortal) to render the tabs container inside the datatable's scroll area.

### CODE REVIEW

File: src/components/mods/modsTable.tsx [hunk 1]
```diff
@@ -31,9 +32,14 @@ const useStyles = createStyles(
     ) => {
         return ({
             tabContainer: {
-                padding: "0 15px",
+                height: "29px",
```
Comment (otobot1) [inline src/components/mods/modsTable.tsx:35]: Does this need to be here? I couldn't figure out what it was doing in my testing, but I didn't want to just remove it without asking.

**Review by otobot1 — COMMENTED** [review]
The table rows scroll up behind the table header and the tabs container now - this will need to be fixed before merging. If we could access the rows directly we could use their height on the page to change their `visibility` to `hidden` or something similar, but I don't think we can...

image

**Comment by otobot1** [comment]
> Thanks for taking this on! I wasn't familiar with `portals` before, but I think they're a great solution. I found one issue that will need to be fixed before merging - the table rows scroll up behind the table header and the tabs container now. If we could access the rows directly we could use their height on the page to change their `visibility` to `hidden` or something similar, but I don't think we can...
> 

Never mind lol. I broke the tab container's background color. I'll fix and then merge after my other comment is addressed.

File: src/components/mods/modsTable.tsx (hunk 1 above)
Comment (ShouvikGhosh2048) [reply src/components/mods/modsTable.tsx]: Yeah the `height` isn't required, I've removed it.

**Comment by ShouvikGhosh2048** [comment]
Does this fix #724?

**Comment by otobot1** [comment]
> Does this fix #724?

Good question! I'll deploy this change and ask Touhoe to check as I don't have access to Safari.

**Pull request merged**